### ISSUE TITLE
datalad create gets freaked out by an empty .git

### ISSUE THREAD

**Issue by leej3** [datalad/datalad#3473]
Raising issue at request of @yarikoptic. The git config warning is just noise...

```
root@778236cea510:/opt/afni# mkdir -p /tmp/test/{.git,afni} && cd /tmp/test/afni && datalad create ds
+ mkdir -p /tmp/test/.git /tmp/test/afni
+ cd /tmp/test/afni
+ datalad create ds
[WARNING] It is highly recommended to configure git first (set both user.name and user.email) before using DataLad. Failed to verify that git is configured: CommandError: command '['git', 'config', 'user.name']' failed with exitcode 1
| Failed to run ['git', 'config', 'user.name'] under None. Exit code=1. out= err= [cmd.py:run:550]CommandError: command '['git', 'config', 'user.email']' failed with exitcode 1
| Failed to run ['git', 'config', 'user.email'] under None. Exit code=1. out= err= [cmd.py:run:550].  Some operations might fail or not perform correctly.
[ERROR  ] /tmp/test [base.py:__init__:168] (InvalidGitRepositoryError)
root@778236cea510:/tmp/test/afni# datalad --version
+ datalad --version
datalad 0.11.4
```

This is true of 0.11.5 too. Not master though.

**Comment by mih** [comment]
This is one of the many things that the 0.12 series does not suffer from. You could install the latest RC.

https://pypi.org/project/datalad/0.12.0rc4/

**Comment by kyleam** [comment]
@leej3:
> Not master though.

@mih:
> This is one of the many things that the 0.12 series does not suffer from.

If I try this recipe on master, it still fails but in a different way.

<details>
<summary>failures</summary>

```sh
#!/bin/sh

set -eu

datalad --version
git --version

cd $(mktemp -dt dl-XXXXXXX)
mkdir .git && mkdir dir && cd dir
datalad --dbg create ds
```

On maint:

```
datalad 0.11.5.dev7
git version 2.21.0
Traceback (most recent call last):
  File "/home/kyle/src/python/venvs/datalad/bin/datalad", line 8, in <module>
    main()
  File "/home/kyle/src/python/datalad/datalad/cmdline/main.py", line 494, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/kyle/src/python/datalad/datalad/interface/base.py", line 628, in call_from_parser
    ret = list(ret)
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 429, in generator_func
    result_renderer, result_xfm, _result_filter, **_kwargs):
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 522, in _process_results
    for res in results:
  File "/home/kyle/src/python/datalad/datalad/distribution/create.py", line 250, in __call__
    on_failure='ignore')
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 492, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 480, in return_func
    results = list(results)
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 429, in generator_func
    result_renderer, result_xfm, _result_filter, **_kwargs):
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 522, in _process_results
    for res in results:
  File "/home/kyle/src/python/datalad/datalad/interface/annotate_paths.py", line 674, in __call__
    result_xfm=None, result_filter=None, return_type='list')
  File "/home/kyle/src/python/datalad/datalad/distribution/dataset.py", line 526, in apply_func
    return f(**kwargs)
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 388, in eval_func
    ds=dataset_arg)
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 374, in _get_procedure_specs
    spec = (ds.config if ds and ds.is_installed()
  File "/home/kyle/src/python/datalad/datalad/distribution/dataset.py", line 347, in config
    if self.repo is None:
  File "/home/kyle/src/python/datalad/datalad/distribution/dataset.py", line 264, in repo
    self._repo.is_with_annex():
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1028, in is_with_annex
    for b in self.get_remote_branches())) or \
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1619, in get_remote_branches
    for remote in self.repo.remotes:
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 807, in repo
    self._repo = self.cmd_call_wrapper(Repo, self.path)
  File "/home/kyle/src/python/datalad/datalad/cmd.py", line 199, in __call__
    return self.call(cmd, *args, **kwargs)
  File "/home/kyle/src/python/datalad/datalad/cmd.py", line 589, in call
    return f(*args, **kwargs)
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 410, in Repo
    return gitpy.Repo(*args, **kwargs)
  File "/home/kyle/src/python/venvs/datalad/lib/python3.5/site-packages/git/repo/base.py", line 168, in __init__
    raise InvalidGitRepositoryError(epath)
git.exc.InvalidGitRepositoryError: /tmp/dl-oygGVlP

> /home/kyle/src/python/venvs/datalad/lib/python3.5/site-packages/git/repo/base.py(168)__init__()
-> raise InvalidGitRepositoryError(epath)
```

On master:
```
datalad 0.12.0rc4.dev108
git version 2.21.0
Traceback (most recent call last):
  File "/home/kyle/src/python/venvs/datalad/bin/datalad", line 8, in <module>
    main()
  File "/home/kyle/src/python/datalad/datalad/cmdline/main.py", line 500, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/kyle/src/python/datalad/datalad/interface/base.py", line 645, in call_from_parser
    ret = list(ret)
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 428, in generator_func
    result_renderer, result_xfm, _result_filter, **_kwargs):
  File "/home/kyle/src/python/datalad/datalad/interface/utils.py", line 520, in _process_results
    for res in results:
  File "/home/kyle/src/python/datalad/datalad/core/local/create.py", line 245, in __call__
    paths=[check_path.relative_to(parentds_path)])
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 3077, in status
    fr='HEAD' if self.get_hexsha() else None,
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1483, in get_hexsha
    commitish)
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1461, in format_commit
    raise e
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1454, in format_commit
    '', cmd, expect_stderr=True, expect_fail=True)
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 327, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1830, in _git_custom_command
    expect_fail=expect_fail)
  File "/home/kyle/src/python/datalad/datalad/support/gitrepo.py", line 1867, in _run_command_files_split
    *args, **kwargs)
  File "/home/kyle/src/python/datalad/datalad/cmd.py", line 694, in run
    cmd, env=self.get_git_environ_adjusted(env), *args, **kwargs)
  File "/home/kyle/src/python/datalad/datalad/cmd.py", line 553, in run
    raise CommandError(text_type(cmd), msg, status, out[0], out[1])
datalad.support.exceptions.CommandError: CommandError: command '['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'show', '-z', '--no-patch', '--format=%H', '--']' failed with exitcode 128
Failed to run ['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'show', '-z', '--no-patch', '--format=%H', '--'] under '/tmp/dl-kxJ65bC'. Exit code=128. out= err=fatal: not a git repository (or any of the parent directories): .git


> /home/kyle/src/python/datalad/datalad/cmd.py(553)run()
-> raise CommandError(text_type(cmd), msg, status, out[0], out[1]
```

</details>

**Comment by mih** [comment]
Arrgh, the truth can be painful.

**Comment by yarikoptic** [comment]
I think if we are to fix it in 0.11.x ideally we need to follow master in throwing the invalid repo exception.  I don't see what else we could do besides informing the user that something is not right

**Comment by kyleam** [comment]
@yarikoptic:

> I think if we are to fix it in 0.11.x ideally we need to follow master in throwing the invalid repo exception.

It's 0.11.x that throws the invalid repo exception.  master raises a CommandError.  With gh-3475, master will ignore an *empty* .git/ upstairs.

> I don't see what else we could do besides informing the user that something is not right

If someone thought it was worthwhile, they could adjust 0.11.x to ignore empty directories like gh-3475 does.  (That change doesn't apply directly to 0.11.x because it relies on pathlib, particularly iterdir().)

But IMO the InvalidGitRepositoryError shown in the initial post is appropriate enough, and we can let this issue be closed with the merge of 3475 into master.
